src/resolved.ts

```diff
@@ -1,6 +1,7 @@
 import { decodePointerFragment, pointerToPath } from '@stoplight/json';
 import { IResolveError } from '@stoplight/json-ref-resolver/types';
 import { Dictionary, ILocation, IRange, JsonPath, Segment } from '@stoplight/types';
+import { DepGraph } from 'dependency-graph';
```

Comment:
Perhaps let's list this as devDependency? We need it for typings and in tests.

Reply:
I'm not sure to follow you.

We also leverage it as the core component to make the whole _"Find me all the orphans!"_ thingie work.

Depending on how users leverage `yarn`, marking it as a devDependency might hinder the overall behavior.

Or did I completely misunderstand you (which wouldn't be the first time and is bound to happen again :wink:)?

![image](https://user-images.githubusercontent.com/92363/66624188-5b1f4d80-ebef-11e9-8f12-c080fed2bdd0.png)

Reply:
> We also leverage it as the core component to make the whole "Find me all the orphans!" thingie work.

Yeah, but jso-ref-resolver already has `dep-graph` listed among its dependencies, as it's required for it to work properly. In general, treat `graph` exposed on resolved object as a public API json-ref-resolver exposes.
In case of Spectral, we need it only for typings.

Reply:
Very clear explanation!

Fixed!